### PR TITLE
Add support for Sonarqube 5.6.x

### PR DESCRIPTION
**Pull request by Dubes**
Thanks for the awesome plugin!

This PR introduces support for generating generic reports in format compatible with Sonarqube 5.6.x which is the current LTS release. 

I have tried to find the balance between minimizing the changes and keeping it open for modification for latest Sonar changes. 

Sonarqube 5.6.x is "off" by default and this PR should not be breaking any existing functionality.

Happy to receive any feedback/criticism :-)

### CODE REVIEW

**Comment by 3dmind** [comment]
Hi Dubes. Thanks for you PR. I will have a closer look.

File: lib/testExecutions.js [hunk 1]
```diff
@@ -2,9 +2,11 @@
 
 const file = require('./file')
 
-module.exports = function testExecutions(data) {
+module.exports = function testExecutions(data, formatForSonar56 = false) {
```
Comment (3dmind) [inline lib/testExecutions.js:5]: Default parameter values are not supported by node 4.
I would like still support node 4 to support a broad basis of node installations.

File: lib/reporter.js [hunk 2]
```diff
@@ -12,13 +12,14 @@ const root = process.cwd()
 const defaultConfig = {
   indent: 2,
   reportPath: process.env.TEST_REPORT_PATH || root,
-  reportFile: 'test-report.xml'
+  reportFile: 'test-report.xml',
+  sonar56x : false
```
Comment (3dmind) [inline lib/reporter.js:16]: I think `sonarLTS` is probably a better name to make the configuration option independent from the actual LTS version number.

**Comment by 3dmind** [comment]
Because of the default parameter value feature, the reporter can not be executed with node 4.
I would use `sonarLTS` over `sonar56x` for configuration options and parameter names to make these names independent of the actual LTS version number.

Would do you think @Dubes?

**Comment by Dubes** [comment]
@3dmind I initially did not choose LTS in the variable name as strictly speaking, this functionality seems to have been introduced in [Sonar 6.2](https://docs.sonarqube.org/display/SONARQUBE62/Generic+Test+Data) so this PR would be required by anyone using sonar 6.1 or lower. 

However, I think it is fair to assume that people who are on Sonar 6 would be using 6.2 or higher since it has been out since December last year. I could not find any mention of plans for Sonarqube 7, I think it is also fair to assume that 5.6.x will remain the LTS version for a while.

Thus **sonarLTS** would be fairer name than sonar56x.

Let me know if you would like me to change it.

**Review by 3dmind — APPROVED** [review]

**Pull request merged**